### ISSUE TITLE
Assigning low memory mox node

### ISSUE THREAD

**Issue by sr320** [RobertsLab/resources#274]
Is there a way to assign a job to a low memory node? For example my current job was low memory however I presume it assigned it to the large memory node as there is one job in the Q.

**Comment by sr320** [comment]
Maybe it was because I used
#SBATCH --time=10-100:00:00
## Memory per node
#SBATCH --mem=128G

Next time will use 127

**Comment by kubu4** [comment]
> 05/22/2018 9:20 AM PDT 
> Hi Sam,
>
> It is actually a bit more complicated than that.  For groups with 3 or fewer nodes, we put your nodes in a "pool" with all of the other groups with 3 or fewer nodes.  This gives you resiliency so you don't lose half of your computing power if a nodes goes down.
>
> Scheduler settings ensure that your partition is entitled to (and limited to) the same resources as if your partition had your two nodes.
> 
> There's a lot of stuff that goes into how the scheduler picks a node, so unless you require a specific node with your job submission (possible, but not recommended), all you can assume ahead of time is that you will get a node with at least the resources you requested available.  All things being equal, the scheduler would pick the node with the minimal resources that meet what you requested for the job in order to leave the nodes with more resources available for "bigger" jobs -- but there are so many factors at play that you can't assume that.
>
> A few things to note:
>
>    In our cluster (with the exception of submissions to the "build" partition and a couple other exception nodes) that all jobs run in "exclusive" mode -- so your job will always be the only job on a node when it runs.
    Regardless of the node your job runs on, the system uses cgroups to limit the memory use to the amount that was requested for a job.  So if you request 96GB, you only get 96GB to use regardless of the node it lands on.
    Because of 1) and 2), it doesn't matter so much what node you actually land on, because you have the same amount of memory to work with either way.
>
> So, in short, the best strategy is to request what your jobs needs to run and let the scheduler worry about the rest.  If you have both "big" and "small" jobs (in terms of resource requirements) to submit at the same time, it certain circumstances it may be advantageous to submit the larger jobs first (that way the scheduler "knows" about the larger job when it schedules the smaller one and can accommodate).
>
>Hopefully that helps somewhat?
>
> -Matt

**Comment by kubu4** [comment]
One last thought on this (for now).

I see that you requested 128GB of RAM. This is the max for our smaller node. In the past, I've requested the full 512GB of RAM for our older node and have had jobs fail due to memory limitations. However, if I submitted the same job, but didn't request the max RAM (e.g. 500GB instead), that same job will run to completion.

So, I'm thinking that even there's a bit of RAM that's not technically available (because some of the RAM is used for basic node operations - sort of like how when you buy a 1TB HDD, but only 975GB is actually available for use). Thus, the new node doesn't _actually_ have 128GB of RAM available for running jobs. Thus, your job submission got pushed to the older node, which has a full 128GB of RAM available.
